config/types/partition.go

```diff
@@ -15,12 +15,22 @@
 package types
 
 import (
+	"errors"
 	"fmt"
 	"regexp"
 
 	"github.com/coreos/ignition/config/validate/report"
 )
 
+const (
+	uuidRegexStr = "^(|[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12})$"
```

Comment:
Looks like the OR is there (assuming you're talking about the third character, the `|`) is necessary to allow empty strings to pass validation, since we don't require users to specify the guid / type guid.